cli/module_graph2.rs

```diff
@@ -560,6 +571,35 @@ pub struct CheckOptions {
   pub reload: bool,
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub enum BundleType {
+  /// Return the emitted contents of the program as a single "flattened" ES
+  /// module.
+  Esm,
+  // TODO(@kitsonk) once available in swc
+  // Iife,
+  /// Do not bundle the emit, instead returning each of the modules that are
+  /// part of the program as individual files.
+  None,
+}
+
+impl Default for BundleType {
+  fn default() -> Self {
+    BundleType::None
+  }
+}
+
+#[derive(Debug, Default)]
+pub struct EmitOptions {
+  /// Indicate the form the result of the emit should take.
+  pub bundle_type: BundleType,
```

Comment:
To be honest I find having both compilation and bundling done via single API configured by this field a little hard to grep, I had already reworked TSC once to split them into separate functions. 

@kitsonk how do you feel about having separate APIs for this purpose? Eg. `ModuleGraph::emit` and `ModuleGraph::emit_bundle`

Reply:
I would prefer not to, because I am of a strong opinion that we should have a single runtime API now of `Deno.emit()` which takes a bundle.  It also makes things easier/cleaner when we drop the emit from `tsc` in lieu of `swc`.

Reply:
Okay, I'll hold off and circle back after getting myself more familiar with the API